src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -105,1002 +98,1361 @@
+        List<MedicalWard> medicalWardList = new ArrayList<>();
+        Medical medical;
+        MedicalInventoryRow inventoryRowTemp;
+        if (code != null) {
+            medical = medicalBrowsingManager.getMedicalByMedicalCode(code);
+            if (medical != null) {
+                medicalWardList = movWardBrowserManager.getMedicalsWard(wardId, medical.getCode(), false);
+            } else {
+                MessageDialog.error(null, MessageBundle.getMessage("angal.inventory.noproductfound.msg"));
+            }
+        } else {
+            medicalWardList = movWardBrowserManager.getMedicalsWard(wardId, false);
+        }
+        for (MedicalWard medicalWard: medicalWardList) {
+            inventoryRowTemp = new MedicalInventoryRow(0, medicalWard.getQty(), medicalWard.getQty(), null,
+                medicalWard.getMedical(), medicalWard.getLot());
+            if (!existInInventorySearchList(inventoryRowTemp)) {
+                inventoryRowsList.add(inventoryRowTemp);
+            }
+        }
+        return inventoryRowsList;
+    }
+
+    private boolean existInInventorySearchList(MedicalInventoryRow inventoryRow) {
+        boolean found = false;
+        List<MedicalInventoryRow> invRows = inventoryRowSearchList.stream()
+                .filter(inventoryRow1 -> inventoryRow1.getMedical().getCode().equals(inventoryRow.getMedical().getCode())).toList();
+        if (!invRows.isEmpty()) {
+            for (MedicalInventoryRow invR: invRows) {
+                if (inventoryRow.getLot() != null && invR.getLot() != null) {
+                    if (inventoryRow.getLot().getCode().equals(invR.getLot().getCode())) {
+                        found = true;
+                        break;
+                    }
+                } else {
+                    if (invR.getLot() == null && inventoryRow.getLot() == null) {
+                        found = true;
+                        break;
+                    }
+                }
+            }
+        }
+        return found;
+    }
+
+    private JRadioButton getSpecificRadio() {
+        if (specificRadio == null) {
+            specificRadio = new JRadioButton(MessageBundle.getMessage("angal.inventory.specificproduct.btn"));
+            specificRadio.addActionListener(actionEvent -> {
+                if (specificRadio.isSelected()) {
+                    codeTextField.setEnabled(true);
+                    codeTextField.setText("");
+                    allRadio.setSelected(false);
+                }
+            });
+        }
+        return specificRadio;
+    }
+
+    private JRadioButton getAllRadio() {
+        if (allRadio == null) {
+            allRadio = new JRadioButton(MessageBundle.getMessage("angal.inventory.allproduct.btn"));
+            allRadio.setSelected(inventory != null);
+            specificRadio.setSelected(inventory == null);
+            allRadio.addActionListener(actionEvent -> {
+                if (!selectAll) {
```

Comment:
When I close the edit inventory window and edit the inventory again, the `selectAll` is again false and by clicking on the `allRadio` button it will add again all products.

![image](https://github.com/user-attachments/assets/bef094a7-f5a3-42a6-839d-614226367c12)

I think we can overcome to all this by really adding only products/lots that are missing in the inventoryRowSearchList instead to rely on a boolean to blindly add them all or not.